napi/src/threadsafe_function.rs

```diff
@@ -277,35 +277,43 @@ impl<T: 'static> ThreadsafeFunction<T, ErrorStrategy::CalleeHandled> {
   /// See [napi_call_threadsafe_function](https://nodejs.org/api/n-api.html#n_api_napi_call_threadsafe_function)
   /// for more information.
   pub fn call(&self, value: Result<T>, mode: ThreadsafeFunctionCallMode) -> Status {
+    self.call_with_result(value, mode).0
+  }
+
+  pub fn call_with_result(
+    &self,
+    value: Result<T>,
+    mode: ThreadsafeFunctionCallMode,
+  ) -> (Status, Result<T>) {
     if self.aborted.load(Ordering::Acquire) {
-      return Status::Closing;
-    }
-    unsafe {
-      sys::napi_call_threadsafe_function(
-        self.raw_tsfn,
-        Box::into_raw(Box::new(value)) as *mut _,
-        mode.into(),
-      )
+      return (Status::Closing, value);
     }
-    .into()
+    let data = Box::into_raw(Box::new(value)) as *mut Result<T>;
+    let status =
+      unsafe { sys::napi_call_threadsafe_function(self.raw_tsfn, data as *mut _, mode.into()) }
```

Comment:
CI failed because the `value: Result<T>` dropped here, and segfault when take it back `Box::from_raw` after in `call_js_cb`